src/Security/Authentication/FrontendPreviewAuthenticator.php

```diff
@@ -68,24 +61,37 @@ public function __construct(RequestStack $requestStack, SessionInterface $sessio
     /**
      * Authenticates a front end user based on the username.
      *
-     * @param null $username
+     * @param string $username
+     *
+     * @return bool
      */
-    public function authenticateFrontendUser($username = null): void
+    public function authenticateFrontendUser(string $username): bool
     {
-        $providerKey = 'contao_frontend';
-        $request = $this->requestStack->getCurrentRequest();
+        $token = $this->tokenStorage->getToken();
 
         // Check if a back end user is authenticated
-        if (null === $this->tokenStorage->getToken() || !$this->tokenStorage->getToken()->isAuthenticated()) {
-            return;
+        if (null === $token || !$token->isAuthenticated()) {
+            return false;
         }
 
-        if (null === $username || !$request->hasSession()) {
-            return;
+        $backendUser = $token->getUser();
+
+        // No back end user logged in or session not started
+        if (!$backendUser instanceof BackendUser || !$this->session->isStarted()) {
+            return false;
+        }
+
+        // Back end user does not have permission to log in front end users
+        if (!$backendUser->isAdmin && (!\is_array($backendUser->amg) && !empty($backendUser->amg))) {
```

Comment:
You should check `empty()` before `is_array()`.

Reply:
Fixed in 6d55b1517aa70f763b07b65bb8ca5e0b9b09e041